File: backend/ws-service/src/main/java/com/example/wsservice/fcm/client/FcmFeignClient.java
package com.example.wsservice.fcm.client;

import com.example.wsservice.fcm.dto.FcmMessageDto;
import com.example.wsservice.fcm.dto.FcmUserMessageDto;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

@FeignClient("push-service")
public interface FcmFeignClient {

    @PostMapping("/api-internal/push/chat")
    Void sendMessage(@RequestBody FcmMessageDto fcmMessageDto);

    @PostMapping("/api-internal/push/user")
    Void sendUserMessage(@RequestBody FcmUserMessageDto fcmUserMessageDto);
}

File: backend/ws-service/src/main/kotlin/com/example/wsservice/fcm/client/FcmFeignClient.kt
package com.example.wsservice.fcm.client

import com.example.wsservice.fcm.dto.FcmMessageDto
import com.example.wsservice.fcm.dto.FcmUserMessageDto
import org.springframework.cloud.openfeign.FeignClient
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody

@FeignClient("push-service")
interface FcmFeignClient {

    @PostMapping("/api-internal/push/chat")
    fun sendMessage(@RequestBody fcmMessageDto: FcmMessageDto): Void?

    @PostMapping("/api-internal/push/user")
    fun sendUserMessage(@RequestBody fcmUserMessageDto: FcmUserMessageDto): Void?
}
